Plutarch/Deriver/PNewtype.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
I put this under `Plutarch.Deriver.PNewType`. If we go with this separate deriver organization - we should probably put the other derivers within `Deriver.*` as well.

If not, let me know where to put this file.

Reply:
We probably want to export this in the prelude TBH.

Reply:
Well the derivers should usually be next to the definition of the class itself. This one deriver is a special-case IMO, so it's probably better to put in `Plutarch.hs` and export from `Plutarch.Prelude` too.

Reply:
Can't put it directly in `Plutarch` due to the dependency on `.Builtin`, just put it in prelude maybe?

Reply:
You'd define `DerivePNewtype` (or whatever name you choose) in `Plutarch`, then have an "orphan" instance in `Plutarch.Builtin`. It's not really an orphan instance because it's impossible to import the class without also importing the instance for `DerivePNewtype`, hence it's 100% safe.

Reply:
Similarly, downstream users who make their own classes, e.g. `PMonad`, would also implement an instance for this type in the same module they define the class.

Reply:
That makes sense, done!